### DISCUSSION TITLE
5.8 并查集

### DISCUSSION THREAD

**Discussion by giscus** [itcharge/AlgoNote#132] [Comments]
# 5.8 并查集

1. 并查集简介 1.1 并查集的定义 并查集（Union Find）：一种树型的数据结构，用于处理一些不交集（Disjoint Sets）的合并及查询问题。不交集指的是一系列没有重复元素的集合。 并查集主要支持两种操作： - 合并（Union）：将两个集合合并成一个集合。 - 查找（Find）：确定某个元素属于哪个集合。通常是返回集合内的一个「代表元...

[https://algo.itcharge.cn/05_tree/05_08_union_find/](https://algo.itcharge.cn/05_tree/05_08_union_find/)

**Comment by HalfCheesee** [comment]
3.1中 按深度合并的代码
union  中最后的 else 
深度相同时，rank[y] += 1
意思是不是 self.rank[y_root] += 1
